### ISSUE TITLE
Sass reads a comment inside a `url()` wherever the parentheses hold no bare address, and the plugin reads an address

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#661]
### What minimal example or steps are needed to reproduce the bug?

Sass reads the parentheses of a `url()` as code wherever what they hold is not a bare address, and a double slash there opens a comment. The plugin reads a bare address wherever no quotation mark stands behind the `(`, so those slashes are characters of the address to it.

```scss
a { b: url(// c
) 1px; }
```

Sass compiles that to `a { b: url() 1px; }` — the comment runs to the end of its line, and the parentheses close on the next. Less compiles it to `a { b: url(// c ) 1px; }`, an address; both PostCSS tokenizers hand it back as one `brackets` token; the plugin records the address `// c`.

Nothing then keeps a fix out of the comment. Measured over every rule and primary option on `9e866cc`, under `postcss-scss`:

```scss
/* function-parentheses-space-inside: never       */ a { b: url(// c) 1px; }
/* function-parentheses-newline-inside: never-multi-line */ a { b: url(// c) 1px; }
/* indentation: tab                                */ a { b: url(// c⏎⇥) 1px; }
/* linebreaks: windows                             */ a { b: url(// c␍⏎) 1px; }
```

The first two take away the line break that closes the comment, and Sass then fails on the output with `expected ")"`, where it compiled the input.

Two more spellings put the same question. `url( a /* ) */ )` is a comment to Sass, which compiles it to `url(a)`, and the address `a /*` to the plugin, so `value-slash-space-before: always` writes `url( a /* ) * / )` and Sass fails with `expected more input`. `url( a(b)c )` is a call of two arguments to Sass, `url(a(b) c)`, and one address to the plugin.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/scss/function-parentheses-space-inside": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

A fix is not written where the compiler the file is written for reads a comment, which is the reading [#557](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/557) settled for the quoted form.

### What actually happened?

The fix writes there, and the stylesheet stops compiling.

### What is the cause?

`readAddress` in `lib/utils/readAddress/index.ts` asks for a quotation mark behind the `(`, whitespace aside, and reads everything else as a bare address running to the first `)`. Sass asks instead whether the text is an unquoted URL at all: a run of whitespace inside it, a parenthesis or a comment makes it none, and the parentheses are then read as code. So `url(a//b)` is an address to both, while `url(// c)`, `url( a /* ) */ )` and `url( a(b)c )` part them.

### Do you have a proposal to fix the bug?

[#557](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/557) settled that the reading a fix is guarded by is the compiler's, and this is the rest of that reading: the parentheses hold a bare address only where the text they hold is one. The narrowest form of it is to read them as code wherever the text holds whitespace with something behind it, a `(` or a `/*`, which is a widening of what the plugin calls a comment and so costs false negatives alone. Measure it with `scripts/sweeps/address-in-a-text.ts`, adding the three spellings above; note that [#660](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/660) asks a narrower version of the same question against PostCSS's tokenizer rather than against Sass, and that Less answers it the other way — it reads `url(// c)` as an address — so the two namespaces may not take one answer.
